aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/RemoveDefaultsDecorator.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rustsdk.customize
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.rust.codegen.client.smithy.ClientRustSettings
+import software.amazon.smithy.rust.codegen.client.smithy.customize.ClientCodegenDecorator
+import software.amazon.smithy.rust.codegen.core.util.shapeId
+import java.util.logging.Logger
+
+/**
+ * Removes default values from certain shapes, and any member that targets those shapes,
+ * for some services where the default value causes serialization issues, validation
+ * issues, or other unexpected behavior.
+ */
+class RemoveDefaultsDecorator : ClientCodegenDecorator {
+    override val name: String = "RemoveDefaults"
+    override val order: Byte = 0
+    private val logger: Logger = Logger.getLogger(javaClass.name)
+    // Service shape id -> Shape id of each root shape to remove the default from.
+    private val removeDefaults = mapOf(
+        "com.amazonaws.emrserverless#AwsToledoWebService".shapeId() to setOf(
+            // Service expects this to have a min value > 0
+            "com.amazonaws.emrserverless#WorkerCounts".shapeId()
```

Comment:
do we want to do it based on the range check? Or is hard coding it the right thing to do here?

Reply:
I thought about doing that. But my reasoning for not doing so was that since the range is wrong now, this will generate an `Option<T>`. Down the road, if the range check became not wrong, it would be a breaking change. So I think its best to just remove the default and always do it to be consistent.